Rick-and-Morty/Rick And Morty/Services/RickAndMortyService.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  RickAndMortyService.swift
+//  Rick And Morty
+//
+//  Created by Scottie Gray on 2021-08-10.
+//  Copyright © 2021 Novoda. All rights reserved.
+//
+
+import Foundation
+
+final class RickAndMortyService: RickAndMortyServiceProtocol {
+    static let baseURL = URL(string: "https://rickandmortyapi.com/api")!
+    
+    func fetchData<T:Decodable>(url: URL, success: @escaping (T) -> (), error: @escaping (Error?) -> ()) {
+        let request = URLRequest(url: url)
+
+        URLSession.shared.dataTask(with: request) { data, response, requestError in
+            if requestError != nil {
+                DispatchQueue.main.async {
+                    error(requestError)
+                }
+            }
```

Comment:
Do we need this here if its being done in the else on line 37? Also why is requestError not being checked for nil in that else?